filter-parser/src/lib.rs

```diff
@@ -100,7 +101,7 @@ impl<'a> Token<'a> {
     }
 
     pub fn parse_finite_float(&self) -> Result<f64, Error> {
-        let value: f64 = self.span.parse().map_err(|e| self.as_external_error(e))?;
+        let value: f64 = self.value().parse().map_err(|e| self.as_external_error(e))?;
```

Comment:
I propagated the change to the original `parse<T>` function that replace span by value() here. This feels like a bug fix indeed, but I am uncertain of its impact. Can you give me some insight as to why this was changed?

Reply:
Thanks for the explanation ☺️ Is this documented somewhere in the code? If not, I feel like it possibly ought to?

Reply:
I added a clear explanation (I think) about the implementation of the geoboundingbox + simplified a little bit the usage of the `Token`.